File: java/src/main/java/com/meteroid/models/SubscriptionActivationConditionEnum.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum SubscriptionActivationConditionEnum implements ToQueryParam {
    ON_START("ON_START"),
    ON_CHECKOUT("ON_CHECKOUT"),
    MANUAL("MANUAL");
    private final String value;

    SubscriptionActivationConditionEnum(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: python/meteroid/models/subscription_activation_condition_enum.py
# this file is @generated
import enum


class SubscriptionActivationConditionEnum(str, enum.Enum):
    ON_START = "ON_START"
    ON_CHECKOUT = "ON_CHECKOUT"
    MANUAL = "MANUAL"

    def __str__(self) -> str:
        return str(self.value)
